client/webserver/passbytes.go

```diff
@@ -0,0 +1,188 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package webserver
+
+import (
+	"fmt"
+	"strconv"
+	"unicode"
+	"unicode/utf16"
+	"unicode/utf8"
+)
+
+const (
+	escapeSequence        = '\\'
+	unicodePrefix         = 'u'
+	unicodeSequenceLength = 6
+)
+
+// PassBytes is an alias of type []byte that implements json.UnmarshalJSON
+// to enable parsing json-encoded strings as []byte.
+type PassBytes []byte
+
+// UnmarshalJSON satisfies the json.Unmarshaler interface, parses json-encoded
+// strings as []byte. Returns an error if the raw bytes to be unmarshalled is
+// not a valid json-encoded string.
+func (pb *PassBytes) UnmarshalJSON(rawBytes []byte) error {
+	rawBytes, err := sanitizeJSONEncodedString(rawBytes)
+	if err != nil {
+		return fmt.Errorf("cannot unmarshal password: %v", err)
+	}
+	*pb = make([]byte, len(rawBytes))
+	copy((*pb)[:], rawBytes)
+	return nil
+}
+
+func (pb PassBytes) String() string {
+	return string(pb)
+}
+
+// sanitizeJSONEncodedString receives a json-encoded string in []byte and
+// returns a new []byte representing the same string but without quotes and
+// escape characters.
```

Comment:
Since it's heavily influenced by [`encoding/json.(*decodeState).unquoteBytes`](https://golang.org/src/encoding/json/decode.go#L1205), with some comments remaining verbatim, I think it's fair and prudent to refer to the std lib function in the docs here although the comments below could still be revised to reflect our understanding.  Although in general I think the rewrite is good and even an improvement in terms of readability.

Reply:
> in general I think the rewrite is good and even an improvement in terms of readability.

I do too.